### PR TITLE
add customer name secondary index to TPCC

### PR DESCRIPTION
**Pull request by johnfangAFW**
Implemented method getCIdByLastNameViaIndex() to replace getCIdByLastName().

TPCC integration tests succeeded.

>>>> Test ./test_k23si_tpcc.sh finished with code 0

### CODE REVIEW

File: src/k2/cmd/tpcc/transactions.h [hunk 1]
```diff
@@ -241,8 +241,72 @@ class PaymentT : public TPCCTxn
...
+                    return make_ready_future();
+                });
+            });
+        });
+    }
+
     // get customer ID by last name
-    // 60% select customer by last name; 40%, the _c_id has already been set randomly at the consturctor
+    // 60% select customer by last name; 40%, the _c_id has already been set randomly at the constructor
     future<> getCIdByLastName() {
```
Comment (jfunston) [inline src/k2/cmd/tpcc/transactions.h:310]: You can remove this function now

File: src/k2/cmd/tpcc/transactions.h [hunk 2]
```diff
@@ -241,8 +241,72 @@ class PaymentT : public TPCCTxn
         return writeRow<History>(history, _txn).discard_result();
     }
 
+    future<> getCIdByLastNameViaIndex() {
+        uint32_t cid_type = _random.UniformRandom(1, 100);
+        if (cid_type > 60) {
```
Comment (jfunston) [inline src/k2/cmd/tpcc/transactions.h:246]: It should be 60% chance to select by last name, so this should be <= 60. This bug existed in the previous code but we might as well fix it now.

File: src/k2/cmd/tpcc/transactions.h [hunk 3]
```diff
@@ -241,8 +241,72 @@ class PaymentT : public TPCCTxn
         return writeRow<History>(history, _txn).discard_result();
     }
 
+    future<> getCIdByLastNameViaIndex() {
```
Comment (jfunston) [inline src/k2/cmd/tpcc/transactions.h:244]: The order status transaction also selects customer by last name and can use the secondary index.

**Review by jfunston — CHANGES_REQUESTED** [review]

File: src/k2/cmd/tpcc/transactions.h (hunk 1 above)
Comment (johnfangAFW) [reply src/k2/cmd/tpcc/transactions.h]: removed, thanks

File: src/k2/cmd/tpcc/transactions.h (hunk 2 above)
Comment (johnfangAFW) [reply src/k2/cmd/tpcc/transactions.h]: fixed and changed it to <=60

File: src/k2/cmd/tpcc/transactions.h (hunk 3 above)
Comment (johnfangAFW) [reply src/k2/cmd/tpcc/transactions.h:238]: replaced getCustomerId() with getCustomerIdViaIndex() for the order status transaction.

**Review by jfunston — APPROVED** [review]

**Pull request merged**